js/my.js

```diff
@@ -200,10 +200,22 @@ document.addEventListener("DOMContentLoaded", function () {
         }
         msg.addText(from, Utf8.decode(text));
 
-        var img = "https://raw.github.com/nickdesaulniers/fxos-irc/master/128.png";
+        var img = "https://raw.githubusercontent.com/nickdesaulniers/fxos-irc/master/images/128.png";
+
         if ($("container").selectedCard.id !== msg.card.id) {
           // This hack is because origin is not supported in manifests for < 1.1.
-          sendNotification(from, { body: Utf8.decode(text), icon: img });
+          var notif = sendNotification(from, { body: Utf8.decode(text), icon: img });
+
+          notif.onclick = function() {
+              var request = window.navigator.mozApps.getSelf();
```

Comment:
We should check `document.visibilityState` first to see if it's "hidden" or "visible" first.  If visible, it will be faster to call showTab than wait for getSelf and lanuch.